### PR TITLE
contribute to #130  Typescript migration

### PR DESCRIPTION
**Pull request by bellaabdelouahab**

### CODE REVIEW

File: backend-app/middlewares/api_version_controll.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { Request, Response, NextFunction } from 'express';
+import { API_VERSION } from '@config/app_config';
+
+/**
+ * Middleware to set default API version in the request URL if not provided.
+ * If the API version is missing, it will be set to 'v3' by default.
+ * @param {Request} req - Express request object.
+ * @param {Response} res - Express response object.
+ * @param {NextFunction} next - Express next function to pass control to the next middleware or route handler.
```
Comment (muttaqin1) [inline backend-app/middlewares/api_version_controll.ts:9]: We dont need this. Cause we are now using ts.

**Review by muttaqin1 — APPROVED** [review]

File: backend-app/middlewares/api_version_controll.ts (hunk 1 above)
Comment (bellaabdelouahab) [reply backend-app/middlewares/api_version_controll.ts:9]: okay 👌

**Comment by GiovanniCaiazzo01** [comment]
@bellaabdelouahab `routes/index.js` has been merged, you can mark it as complete

**Comment by bellaabdelouahab** [comment]
@GiovanniCaiazzo01 done

**Comment by bellaabdelouahab** [comment]
@muttaqin1 @Khalid1G 
give your final review

**Comment by muttaqin1** [comment]
> @muttaqin1 @Khalid1G 
> give your final review

Sure, i will thoroughly review it soon.

**Comment by bellaabdelouahab** [comment]
- Resolves #129

**Comment by bellaabdelouahab** [comment]
- Resolves #130

**Comment by bellaabdelouahab** [comment]
@muttaqin1 by the way to review this in the right way you can check out to the pr and test it directly, because i see lots of people reviewing just by reading the files and not running them

**Comment by muttaqin1** [comment]
> @muttaqin1 by the way to review this in the right way you can check out to the pr and test it directly, because i see lots of people reviewing just by reading the files and not running them 

Okay.

File: backend-app/app.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+app.use('*', (req: Request, _res: Response, next: NextFunction) => {
+    const err = new AppError(404, 'Route Not Found', req.originalUrl);
+    next(err);
+});
+
+// dummy error handler
+app.use((_req: Request, _res: Response, next: NextFunction) => {
+    const err = new AppError(500, 'Something went wrong');
+    next(err);
+});
```
Comment (muttaqin1) [inline backend-app/app.ts:100]: I think you forgot to remove this.

**Review by muttaqin1 — APPROVED** [review]

File: backend-app/app.ts (hunk 2 above)
Comment (bellaabdelouahab) [reply backend-app/app.ts:100]: oki

**Pull request merged**